test/controllers/audits.test.js

```diff
@@ -926,30 +906,64 @@ describe('Audits Controller', () => {
         expect(result.status).to.equal(200);
       });
 
-      it('successful update if groupedURLs is undefined in the config', async () => {
-        const groupedURLsConfig = undefined;
-        const handlerConfig = {
-          groupedURLs: groupedURLsConfig,
-        };
-        const groupedURLs = [
+      it('successful update if groupedURLs is undefined in the site config', async () => {
+        const currentGroupedURLs = undefined;
+        const requestGroupedURLs = [
           { name: 'catalog', pattern: '/products/' },
           { name: 'blog', pattern: '/posts/' },
         ];
+
         const context = {
-          params: { siteId, auditType: 'broken-backlinks' },
-          data: { groupedURLs },
+          params: { siteId, auditType },
+          data: { groupedURLs: requestGroupedURLs },
         };
 
-        siteConfig.getHandlerConfig.withArgs(auditType).returns(handlerConfig);
-        siteConfig.getGroupedURLs.withArgs(auditType).returns(groupedURLsConfig);
+        siteConfig.getHandlerConfig.withArgs(auditType);
+        siteConfig.getGroupedURLs.withArgs(auditType).returns(currentGroupedURLs);
 
         const result = await auditsController.patchAuditForSite(context);
 
-        expect(siteConfig.updateGroupedURLs.calledWith(auditType, groupedURLs)).to.be.true;
+        expect(siteConfig.updateGroupedURLs.getCall(0).args[0]).to.be.equal(auditType);
+        expect(siteConfig.updateGroupedURLs.getCall(0).args[1]).to.deep.equal(requestGroupedURLs);
         expect(site.updateConfig.calledWith(sinon.match.any)).to.be.true;
         expect(mockDataAccess.updateSite.calledWith(site)).to.be.true;
         expect(result.status).to.equal(200);
       });
+
+      it(
+        'updates existing group instead of creating a new one for duplicate pattern',
+        async () => {
+          const currentGroupedURLs = [
+            { name: 'page', pattern: '/page/*' },
+            { name: 'existed_group', pattern: '/pattern/*' },
+          ];
+          const requestGroupedURLs = [
+            { name: 'group_with_the_same_pattern', pattern: '/pattern/*' },
```

Comment:
could you change value of the name property of the `requestGroupedURLs` so that the test asserts if the final group is the one comes from request, not current